tests/DependencyLoaderTest.php

```diff
@@ -6,28 +6,55 @@
 
 use Dominikb\ComposerLicenseChecker\Contracts\DependencyParser;
 use Dominikb\ComposerLicenseChecker\DependencyLoader;
+use Dominikb\ComposerLicenseChecker\Exceptions\CommandExecutionException;
 use Mockery;
+use Symfony\Component\Console\Command\Command;
 
 class DependencyLoaderTest extends TestCase
 {
-    /** @test */
+    /**
+     * @test
+     *
+     * @requires OS Linux
+     * Linux required because of escape characters in the verified command
+     */
     public function it_runs_the_command_with_the_given_inputs()
     {
         $loader = Mockery::mock(DependencyLoader::class, [$this->createNoOpParser()])
                          ->makePartial();
 
         $command = '';
         $loader->shouldAllowMockingProtectedMethods()
-               ->shouldReceive('exec')
-               ->once()
+               ->expects('exec')
                ->withArgs(function ($c) use (&$command) {
                    return (bool) ($command = $c);
                })
-               ->andReturn([]);
+               ->andReturns([]);
 
         $loader->loadDependencies('./composerpath/composer-binary', '/some/directory');
 
-        $this->assertEquals('./composerpath/composer-binary licenses -f json -d /some/directory', $command);
+        $this->assertEquals('\'./composerpath/composer-binary\' licenses --format json --working-dir \'/some/directory\'', $command);
+    }
+
+    /**
+     * @test
+     */
+    public function it_throws_on_exec_failure()
+    {
+        $loader = Mockery::mock(DependencyLoader::class, [$this->createNoOpParser()])
+            ->makePartial();
+
+        $command = '';
+        $loader->shouldAllowMockingProtectedMethods()
+            ->expects('exec')
+            ->withArgs(function ($c) use (&$command) {
+                return (bool) ($command = $c);
+            })
+            ->andThrows(CommandExecutionException::class, 'Error when trying to fetch licenses from Composer', Command::INVALID);
```

Comment:
We can drop unused arguments and lines of code
```suggestion
        $loader->shouldAllowMockingProtectedMethods()
            ->expects('exec')
            ->andThrows(CommandExecutionException::class, 'Error when trying to fetch licenses from Composer', Command::INVALID);
```